ddtrace/contrib/pytest/plugin.py

```diff
@@ -331,94 +336,103 @@ def pytest_runtest_protocol(item, nextitem):
-            span.set_tag_str(_EVENT_TYPE, SpanTypes.TEST)
-            span.set_tag_str(test.NAME, item.name)
-            span.set_tag_str(test.COMMAND, _get_pytest_command(item.config))
-            span.set_tag_str(_SESSION_ID, str(test_session_span.span_id))
-
-            if test_module_span is not None:
-                span.set_tag_str(_MODULE_ID, str(test_module_span.span_id))
-                span.set_tag_str(test.MODULE, test_module_span.get_tag(test.MODULE))
-                span.set_tag_str(test.MODULE_PATH, test_module_span.get_tag(test.MODULE_PATH))
-
-            if test_suite_span is not None:
-                span.set_tag_str(_SUITE_ID, str(test_suite_span.span_id))
-                test_class_hierarchy = _get_test_class_hierarchy(item)
-                if test_class_hierarchy:
-                    span.set_tag_str(test.CLASS_HIERARCHY, test_class_hierarchy)
-                if hasattr(item, "dtest") and isinstance(item.dtest, DocTest):
-                    span.set_tag_str(test.SUITE, "{}.py".format(item.dtest.globs["__name__"]))
-                else:
-                    span.set_tag_str(test.SUITE, test_suite_span.get_tag(test.SUITE))
-
-            span.set_tag_str(test.TYPE, SpanTypes.TEST)
-            span.set_tag_str(test.FRAMEWORK_VERSION, pytest.__version__)
-
-            if item.location and item.location[0]:
-                _CIVisibility.set_codeowners_of(item.location[0], span=span)
-
-            # We preemptively set FAIL as a status, because if pytest_runtest_makereport is not called
-            # (where the actual test status is set), it means there was a pytest error
-            span.set_tag_str(test.STATUS, test.Status.FAIL.value)
-
-            # Parameterized test cases will have a `callspec` attribute attached to the pytest Item object.
-            # Pytest docs: https://docs.pytest.org/en/6.2.x/reference.html#pytest.Function
-            if getattr(item, "callspec", None):
-                parameters = {"arguments": {}, "metadata": {}}  # type: Dict[str, Dict[str, str]]
-                for param_name, param_val in item.callspec.params.items():
-                    try:
-                        parameters["arguments"][param_name] = encode_test_parameter(param_val)
-                    except Exception:
-                        parameters["arguments"][param_name] = "Could not encode"
-                        log.warning("Failed to encode %r", param_name, exc_info=True)
-                span.set_tag_str(test.PARAMETERS, json.dumps(parameters))
-
-            markers = [marker.kwargs for marker in item.iter_markers(name="dd_tags")]
-            for tags in markers:
-                span.set_tags(tags)
-            _store_span(item, span)
-
-            # Run the actual test
-            yield
+        # Finish coverage for the test suite if coverage is enabled
+        if TEST_SKIPPING_LEVEL == TEST and coverage_enabled() and not is_skipped_by_itr:
+            _coverage_end(test_suite_span)
```

Comment:
Why does this end the test suite span's coverage? Wouldn't https://github.com/DataDog/dd-trace-py/blob/abbdd571f48cfddabab298c3efb9600df8516589/ddtrace/internal/ci_visibility/coverage.py#L75 cause the next test to overwrite the coverage tag?